pkg/preflight/preflight.go

```diff
@@ -262,20 +285,35 @@ func GetPreflightCommand(appSlug string) []string {
 	return comamnd
 }
 
-func CreateRenderedSpec(app *apptypes.App, sequence int64, origin string, inCluster bool, kotsKinds *kotsutil.KotsKinds) error {
-	builtPreflight := kotsKinds.Preflight.DeepCopy()
-	if builtPreflight == nil {
-		builtPreflight = &troubleshootv1beta2.Preflight{
-			TypeMeta: v1.TypeMeta{
-				Kind:       "Preflight",
-				APIVersion: "troubleshoot.sh/v1beta2",
-			},
-			ObjectMeta: v1.ObjectMeta{
-				Name: "default-preflight",
-			},
+func CreateRenderedSpec(app *apptypes.App, sequence int64, origin string, inCluster bool, kotsKinds *kotsutil.KotsKinds, archiveDir string) error {
+	tsKinds, err := kotsutil.LoadTSKindsFromPath(filepath.Join(archiveDir, "rendered"))
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("failed to load troubleshoot kinds from path: %s", filepath.Join(archiveDir, "rendered")))
+	}
+
+	builtPreflight := &troubleshootv1beta2.Preflight{}
+	if tsKinds.PreflightsV1Beta2 != nil {
+		for _, v := range tsKinds.PreflightsV1Beta2 {
+			builtPreflight = troubleshootpreflight.ConcatPreflightSpec(builtPreflight, &v)
+		}
+	} else {
+		builtPreflight = kotsKinds.Preflight.DeepCopy()
+		if builtPreflight == nil {
+			builtPreflight = &troubleshootv1beta2.Preflight{
```

Comment:
initializing with the default spec should probably be moved outside of this if/else block.